src/strongbox/addon.clj

```diff
@@ -116,18 +117,27 @@
         addon-list (apply conj (mapv expand addon-groups) unknown-grouping)]
     addon-list))
 
-(defn-spec ungroup-addon :addon/installed-list
-  "an addon may actually be many addons bundled together with a primary one chosen to represent them.
-  sometimes we want to treat this addon as a list of addons"
-  [addon :addon/installed]
-  (if (empty? (:group-addons addon))
-    [addon]
-    (get addon :group-addons [])))
+(defn merge-source-map-lists
+  "merges two lists of source-maps returning a single distinct one or nil if empty"
+  [list-a, list-b]
+  (some->> (into list-a list-b)
+           (remove nil?)
+           utils/nilable
+           distinct
+           vec))
+
+(defn extract-source-map-list
+  "extracts a source-map from the given `data` as well as anything in :source-map-list and returns a single distinct source map list"
+  [data]
+  (merge-source-map-lists
+   (or (some-> data (select-keys [:source :source-id]) utils/nilable vector) [])
+   (get data :source-map-list [])))
 
 (defn merge-toc-nfo
-  "it's own function because the logic is duplicated in tests otherwise"
   [toc nfo]
-  (merge toc nfo))
+  (merge toc nfo (some->> (merge-source-map-lists (extract-source-map-list toc)
```

Comment:
pick this new map out into a `let` binding

docstr

defn-spec

update tests